graylog2-server/src/main/java/org/graylog2/inputs/InputServiceImpl.java

```diff
@@ -142,6 +142,9 @@ public Input create(Map<String, Object> fields) {
 
     @Override
     public Input find(String id) throws NotFoundException {
+        if (!ObjectId.isValid(id)) {
+            throw new NotFoundException("Input id <" + id + "> is invalid!");
```

Comment:
But even then I don't like the mix of different exception semantics here. An invalid ID is definitively not "not found" but simply an invalid ID for which there is the `IllegalArgumentException` in Java. Even the message of the `NotFoundException` in your changeset says that.

I still think this should be an IAE (which then could be automatically mapped to an HTTP 400 response using a JAX-RS exception mapper).